lib/foreman_ansible_core/runner/ansible_runner.rb

```diff
@@ -106,6 +108,22 @@ def write_playbook
         File.write(File.join(@root, 'project', 'playbook.yml'), @playbook)
       end
 
+      def write_ssh_key
+        key_path = File.join(@root, 'env', 'ssh_key')
+        File.symlink(ForemanRemoteExecutionCore.settings[:ssh_identity_key_file], key_path)
```

Comment:
Key file can be specified on a per-host basis.

If I'm reading https://ansible-runner.readthedocs.io/en/stable/intro.html#env-ssh-key right, then ansible-runner doesn't support providing more keys through the env directory and putting a key there overrides the per-host settings. That would also mean this would break the different keys for different hosts use case completely, no matter if a passphrase is used or not.